fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/implementation/NestLoopJoinImplementationRule.java

```diff
@@ -22,6 +24,15 @@ public static NestLoopJoinImplementationRule getInstance() {
         return instance;
     }
 
+    @Override
+    public boolean check(final OptExpression input, OptimizerContext context) {
+        LogicalJoinOperator joinOperator = (LogicalJoinOperator) input.getOp();
+        JoinOperator joinType = joinOperator.getJoinType();
+        ScalarOperator predicate = joinOperator.getOnPredicate();
+        // TODO: support other join types
+        return (joinType.isCrossJoin() || joinType.isInnerJoin() || joinType.isOuterJoin());
```

Comment:
I think we should try not choose to nestloop unless there is a clear performance advantage or feature support, it's will produce many meaningless plans

Reply:
Could our optimizer fragment support this specialization?

Reply:
done